### DISCUSSION TITLE
Is there a recommended method to lint argo-cd yaml files?

### DISCUSSION THREAD

**Discussion by asfaltboy** [argoproj/argo-cd#14407] [Q&A]
We use a pattern of storing all our argo-cd files in a root app in VCS.

We'd like to make sure the files in this app are valid, as it's too easy to introduce yaml typos.

Currently, we run them through a tool like `yq` and check for failures.
And, I noticed that argo workflows has a `argo lint` tool.

Is there something similar for Argo CD? Or, is there another recommended method people like to use?

**Comment by pfarikrispy** [comment]
I am looking for the same thing. Tried `argo lint` but it won't recognize the YAMLs.

I did find golangci-lint and trunk.io but they don't specifically lint argocd files, which is what I'm interested in.

**Comment by rumstead** [comment]
I have implemented a mixture of linting. We use [Conftest](https://www.conftest.dev/) for some business logic like using ssh over https for our git repos. We use `yamllint` for general yaml validation and [kubeconform/kubeeval](https://github.com/yannh/kubeconform?tab=readme-ov-file#CustomResourceDefinition-CRD-Support) for CR linting. It would be great to have this built in to make it easier :).

I originally had a go binary we built with the argo types pulled in and unmarshaled the yamls into structs for validation.

**Comment by pfarikrispy** [comment]
Thanks so much for the suggestions!

We also use yamllint but I didn't know about kubeconform yet. Very useful!
Appreciated!
Christian.

**Comment by jumbosushi** [comment]
For someone finding this much later like me: we were able to enforce a lint pretty easily by using Cue
https://cuelang.org/docs/concept/how-cue-works-with-go/
e.g. Used this struct from ArgoCD Go type: https://github.com/argoproj/argo-cd/blob/v2.14.11/pkg/apis/application/v1alpha1/applicationset_types.go#L63-L75
